File: Support/Templates/cs/Files/EditableChildBindingList.cs
using System;
using System.Collections.Generic;
using Csla;

namespace Templates
{
  [Serializable]
  public class EditableChildBindingList :
    BusinessBindingListBase<EditableChildBindingList, EditableChild>
  {
    [FetchChild]
    private void Fetch(object childData)
    {
      RaiseListChangedEvents = false;
      foreach (var child in (IList<object>)childData)
        this.Add(DataPortal.FetchChild<EditableChild>(child));
      RaiseListChangedEvents = true;
    }
  }
}

File: Support/Templates/vb/Files/EditableChildBindingList.vb
Imports Csla.Security

<Serializable()> _
Public Class EditableChildBindingList
  Inherits BusinessBindingListBase(Of EditableChildBindingList, EditableChild)

  <FetchChild>
  Private Sub Fetch(ByVal childData As Object)
    RaiseListChangedEvents = False
    For Each child As Object In CType(childData, IList(Of Object))
      Me.Add(DataPortal.FetchChild(Of EditableChild)(child))
    Next
    RaiseListChangedEvents = True
  End Sub

End Class
